### PR TITLE
tools: add nerdctl as a new container engine

### PR DESCRIPTION
**Pull request by changweige**
For users that do not have docker installed locally, nerdctl which is a containerd community client is very convenient, especially for containerd and its sub-projects developers.

In addition, try to format the rootfs.sh script making it neat.

Signed-off-by: Changwei Ge <gechangwei@bytedance.com>

### CODE REVIEW

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 1]
```diff
@@ -371,6 +368,8 @@ build_rootfs_distro()
 		elif [ -n "${USE_PODMAN}" ]; then
 			container_engine="podman"
 			engine_build_args+=" --runtime ${DOCKER_RUNTIME}"
+		elif [ -n "${USE_NERDCTL}" ]; then
```
Comment (liubin) [inline tools/osbuilder/rootfs-builder/rootfs.sh:371]: Please can you add `USE_NERDCTl` in `tools/osbuilder/README.md` about how to use it and the use case, for example, combinate it with buildkit.

Comment (liubin) [reply tools/osbuilder/rootfs-builder/rootfs.sh:371]: And please add an option description to the help message https://github.com/kata-containers/kata-containers/blob/f1381eb361aa2202d48607df8cfc6dbfe6db580c/tools/osbuilder/rootfs-builder/rootfs.sh#L151-L157

Comment (changweige) [reply tools/osbuilder/rootfs-builder/rootfs.sh:371]: Sure. will add more information and an introduction for related docs within this PR. Besides, I feel it's better to split this PR into two one of which is only formatting the script.

Comment (amshinde) [reply tools/osbuilder/rootfs-builder/rootfs.sh:371]: Yes please, this should be split into two separate commits. Please include all the formatting changes in a separate commit.

**Comment by changweige** [comment]
Depends on #6255

**Pull request closed without merging**